nsdb-core/src/main/scala/io/radicalbit/nsdb/actors/PublisherActor.scala

```diff
@@ -138,27 +178,77 @@ class PublisherActor(readCoordinator: ActorRef) extends ActorPathLogging {
+                RecordsPublished(id, q.query.metric, Seq.empty)
+            }
+          subscribedActorsByQueryId.get(id).foreach(e => e.foreach(f.pipeTo(_)))
+        case _ => //do nothing
+      }
+
+    case PushTemporalAggregatedQueries(quid) =>
+      for {
+        temporalQuery  <- temporalAggregatedQueries.get(quid)
+        partialResults <- temporalBuckets.get(quid)
+        schema         <- schemas.get(temporalQuery.query.metric)
+      } yield {
+        temporalQuery.parsedQuery match {
+          case parsedTemporalAggregatedQuery: ParsedTemporalAggregatedQuery =>
+            reduceSingleTemporalBucket(schema, temporalQuery.query, parsedTemporalAggregatedQuery.aggregation)(
+              mathContext)(partialResults) match {
+              case Some(record) =>
+                subscribedActorsByQueryId
+                  .get(quid)
+                  .foreach(e => e.foreach(_ ! RecordsPublished(quid, schema.metric, Seq(record))))
+                temporalBuckets -= quid
+              case None => //do nothing
+            }
+          case _ =>
+            log.warning(s"trying to process a non temporal query ${temporalQuery.query} as temporal query")
+        }
+      }
+
     case PublishRecord(db, namespace, metric, record, schema) =>
-      plainQueries.foreach {
-        case (id, nsdbQuery) if nsdbQuery.query.metric == metric && subscribedActorsByQueryId.contains(id) =>
+      (plainQueries ++ temporalAggregatedQueries).foreach {
+        case (quid, nsdbQuery) if nsdbQuery.query.metric == metric && subscribedActorsByQueryId.contains(quid) =>
           implicit val timeContext: TimeContext = TimeContext()
+
+          val temporaryIndex: TemporaryIndex = new TemporaryIndex()
+          implicit val writer: IndexWriter   = temporaryIndex.getWriter
+          temporaryIndex.write(record)
+          writer.close()
+
           nsdbQuery.parsedQuery match {
             case parsedQuery: ParsedSimpleQuery =>
-              val temporaryIndex: TemporaryIndex = new TemporaryIndex()
-              implicit val writer: IndexWriter   = temporaryIndex.getWriter
-              temporaryIndex.write(record)
-              writer.close()
               if (db == nsdbQuery.query.db && namespace == nsdbQuery.query.namespace && metric == nsdbQuery.query.metric && temporaryIndex
                     .query(schema, parsedQuery.q, parsedQuery.fields, 1, None)
                     .lengthCompare(1) == 0)
                 subscribedActorsByQueryId
-                  .get(id)
-                  .foreach(e => e.foreach(_ ! RecordsPublished(id, metric, Seq(record))))
+                  .get(quid)
+                  .foreach(e => e.foreach(_ ! RecordsPublished(quid, metric, Seq(record))))
               temporaryIndex.close()
+            case parsedQuery: ParsedTemporalAggregatedQuery =>
+              if (db == nsdbQuery.query.db && namespace == nsdbQuery.query.namespace && metric == nsdbQuery.query.metric && temporaryIndex
```

Comment:
for readibilty, is it possible to put the if condition in a function and give a meaningful name?

Reply:
Agreed